src/main.lua

```diff
@@ -809,7 +870,7 @@ addEventingHandler(ActionMap.JoinNetwork, utils.hasMatchingTag("Action", ActionM
 	local timestamp = tonumber(msg.Timestamp)
 
 	msg.ioEvent:addField("Resolved-Observer-Address", formattedObserverAddress)
-	msg.ioEvent:addField("Sender-Previous-Balance", fromBalance)
+	msg.ioEvent:addField("Sender-Previous-Balance", Balances[fromAddress] or 0)
```

Comment:
nit - 
```suggestion
	msg.ioEvent:addField("Sender-Previous-Balance", balances.getBalance(fromAddress))
```

Reply:
My issue with that approach is it makes a whole deep copy of the balances table to then just throw it away. Why do that here for a scalar value?

Reply:
Perhaps what we need is a cheaper getter function if you want to protect accesses of the main Balances object?